modules/mapbox/src/deck-utils.ts

```diff
@@ -60,7 +60,15 @@ export function getDeckInstance({
     });
     // If using the WebGLContext created by deck (React use case), we use deck's viewState to drive the map.
     // Otherwise (pure JS use case), we use the map's viewState to drive deck.
-    map.on('move', () => onMapMove(deckInstance, map));
+    const _handleMapMove = () => {
+      // deregister self if called after deck is finalized.
+      if (!deckInstance?.layerManager) {
```

Comment:
This works, however I'd prefer to implement a `deck.isFinalized` since `layerManager` is a protected variable.

Reply:
`deck.isInitialized` basically does this. However it may be `false` in two cases:
- After Deck is constructed and before context become available
- After Deck is finalized

Reply:
Can I add an `isFinalized` that is only true when deck has been finalized?

Reply:
I think we need to discuss how `isInitialized` should behave if we add more tags... Alternatively you can do this

```js
if (deck.isInitialized) {
  watchMapMove(map, deck);
} else {
  deckProps.onLoad = () => watchMapMove(map, deck)
}
```